plugins/inputs/netdata/netdata.go

```diff
@@ -0,0 +1,291 @@
+package netdata
+
+import (
+	"bytes"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+const chartsURL = "/api/v1/charts"
+
+const description = `Read metrics from netdata`
+
+const sampleConfig = `
+  ## Number of points to gather.
+  ## By default inputs plugin run every 10 seconds, by asking 10 points
+  ## to netdata we get measurement for each second. This value is capped by "update every" config of netdata.
+  ## For example you can set
+  ## points = 20
+  ## interval = "10m"
+  ## to get measurement every 30 seconds (and polling netdata every 10 minutes)
+  ## default to 10
+  points = 10
+
+  ## Grouping method if multiple collected values are to be grouped in order to return fewer points.
+  ## methods supported "min", "max", "average", "sum", "incremental-sum"
+  ## cf : https://github.com/firehol/netdata/wiki/REST-API-v1
+  ## default to average
+  group = "average"
+
+  ## Should measurement be prefixed by the netdata hostname ?
+  ## Ex : netdata.host1.apps.cpu (with hostname),  netdata.apps.cpu (without hostname)
+  ## default to true
+  prefixChartIDByHostname = true
+
+  ## Netdata servers
+  [[inputs.netdata.servers]]
+    url="http://netdata-host:19999"
+
+  [[inputs.netdata.servers]]
+    url="https://netdata-host2"
+    basicAuthUsername="user"
+    basicAuthPassword="pass"
+`
+
+type Netdata struct {
+	Servers                 []Server
+	Points                  int
+	Group                   string
+	PrefixChartIDByHostname bool
+
+	client *http.Client
+}
+
+type Server struct {
+	Url               string
+	BasicAuthUsername string
+	BasicAuthPassword string
+	lastGatherTime    int64
+	currGatherTime    int64
+}
+
+type NestedError struct {
+	Err       error
+	NestedErr error
+}
+
+func (ne NestedError) Error() string {
+	return ne.Err.Error() + ": " + ne.NestedErr.Error()
+}
+
+func Errorf(err error, msg string, format ...interface{}) error {
+	return NestedError{
+		NestedErr: err,
+		Err:       fmt.Errorf(msg, format...),
+	}
+}
+
+type ResponseCharts struct {
+	Hostname string           `json:"hostname"`
+	Charts   map[string]Chart `json:"charts"`
+}
+
+type Chart struct {
+	ID         string                    `json:"id"`
+	Type       string                    `json:"type"`
+	Family     string                    `json:"family"`
+	DataURL    string                    `json:"data_url"`
+	Dimensions map[string]ChartDimension `json:"dimensions"`
+}
+
+type ChartDimension struct {
+	Name string `json:"name"`
+}
+
+type ResponseData struct {
+	DimensionNames []string   `json:"dimension_names"`
+	Result         DataResult `json:"result"`
+}
+
+type DataResult struct {
+	Data [][]interface{} `json:"data"`
+}
+
+func init() {
```

Comment:
The `init()` function is usually placed at the end of the plugin code. While not a rule it's kind of a convention. :-)